public/js/modules/nav.js

```diff
@@ -0,0 +1,123 @@
+import {
+  disableBodyScroll,
+  enableBodyScroll,
+} from "../lib/bodyScrollLock.esm.js";
+
+export default class nav {
+  constructor(el) {
+    this.el = el;
+    this.setVars();
+    this.bindEvents();
+  }
+
+  mql = window.matchMedia("(min-width: 768px)");
+
+  get mobileOpen() {
+    return this.el.dataset.mobile === "true";
+  }
+
+  setVars() {
+    // grab targets
+    this.navTarget = this.el.querySelector("[data-nav-target]");
+    this.navToggle = this.el.querySelector("[data-nav-toggle]");
+
+    // build list for focus trap
+    this.focusableEls = this.el.querySelectorAll(
+      'a[href]:not([disabled]), a[href]:not(.skip-link), button:not([disabled]), textarea:not([disabled]), input[type="text"]:not([disabled]), input[type="radio"]:not([disabled]), input[type="checkbox"]:not([disabled]), select:not([disabled])'
+    );
+    this.focusFirst = this.focusableEls[0]
+    this.focusLast = this.focusableEls[this.focusableEls.length - 1]
+
+    // close mobile nav if left open and resized
+    this.resizeObserver = new ResizeObserver(() => {
+      if (this.mql.matches && this.mobileOpen) {
+        console.log('resize close')
+        this.toggleVisibilty();
+      }
+    });
+
+    // track when the sticky nav is pinned
+    this.intersectionObserver = new IntersectionObserver(
+      ([e]) => {
+        setTimeout(() => {
+          e.target.classList.toggle("-pinned", e.intersectionRatio < 1);
+        }, 250);
+      },
+      { threshold: 1 }
+    );
+  }
+
+  bindEvents() {
+    this.navToggle.addEventListener("click", this.toggleVisibilty.bind(this));
+    this.resizeObserver.observe(this.el);
+    this.intersectionObserver.observe(this.el);
+    this.navTarget.addEventListener("transitionend", this.toggleOpacity.bind(this));
+    window.addEventListener('keydown', this.handleKeyDown)
+  }
+
+  toggleVisibilty = () => {
+    if (this.mobileOpen) {
+      // close
+      this.navTarget.classList.remove("-active");
+      this.el.dataset.mobile = false;
+      enableBodyScroll(this.navTarget);
+      document.activeElement.blur();
+      this.el.removeEventListener('keydown', this.focusTrap)
+    } else {
+      // open
+      this.navToggle.ariaPressed = true;
+      this.el.dataset.mobile = true;
+      disableBodyScroll(this.navTarget);
+      this.navTarget.classList.add("-visible");
+      setTimeout(() => {
+        this.navTarget.classList.add("-active");
+      }, 0);
```

Comment:
Feels a little kludgy/could be improved but sets visibility/display before triggering the css transition. The hiding happens after the `transitionend` event on close.